addons/sale_stock/tests/test_sale_stock.py

```diff
@@ -2194,3 +2194,35 @@ def test_multicompany_transit_with_one_company_for_user(self):
         intercom_location = self.env.ref('stock.stock_location_inter_company')
         self.assertEqual(so.picking_ids.location_dest_id, intercom_location)
         self.assertEqual(so.picking_ids.move_ids.location_dest_id, intercom_location)
+
+    def test_cancelled_backorder_product_forecast_widget(self):
+        '''
+        Ensure the availability widget is hidden if back order was cancelled.
+        '''
+        storable_product = self.env['product.product'].create({
+            'name': 'storable_product',
+            'is_storable': True,
+        })
+        admin = self.env['res.users'].search([('login', '=', 'admin')], limit=1)
+        # Use admin because Accountman doesn't have necessary rights
+        self.env["stock.quant"].with_user(admin).create({
+            "product_id": storable_product.id,
+            "location_id": self.env.ref('stock.stock_location_stock').id,
+            "inventory_quantity": 15.0,
+        }).action_apply_inventory()
+
+        sale_order = self.env['sale.order'].create({
+            'partner_id': self.partner_a.id,
+            'order_line': [
+                (0, 0, {'product_id': storable_product.id, 'product_uom_qty': 5, 'sequence': 42}),
+            ],
+        })
+        sale_order.action_confirm()
+        self.assertTrue(sale_order.order_line.display_qty_widget)
+        picking = sale_order.picking_ids
+        partial_move = picking.move_ids[0]
+        partial_move.quantity = 2
```

Comment:
don't assign variable if used only once
```suggestion
        picking.move_ids.quantity = 2
```